src/repository.c

```diff
@@ -565,6 +565,79 @@ Repository_merge_base(Repository *self, PyObject *args)
     return git_oid_to_python(&oid);
 }
 
+typedef int (*git_merge_base_xxx_t)(git_oid *out, git_repository *repo, size_t length, const git_oid input_array[]);
+
+static PyObject *
+merge_base_xxx(Repository *self, PyObject *args, git_merge_base_xxx_t git_merge_base_xxx)
+{
+    PyObject *py_result = NULL;
+    PyObject *py_commit_oid;
+    PyObject *py_commit_oids;
+    git_oid oid;
+    int commit_oid_count;
+    git_oid *commit_oids = NULL;
+    int i = 0;
+    int err;
+
+    if (!PyArg_ParseTuple(args, "O!", &PyList_Type, &py_commit_oids))
+        return NULL;
+
+    commit_oid_count = (int)PyList_Size(py_commit_oids);
+    commit_oids = malloc(commit_oid_count * sizeof(git_oid));
+    if (commit_oids == NULL) {
+        PyErr_SetNone(PyExc_MemoryError);
+        goto out;
+    }
+
+    for (; i < commit_oid_count; i++) {
+        py_commit_oid = PyList_GET_ITEM(py_commit_oids, i);
+        err = py_oid_to_git_oid_expand(self->repo, py_commit_oid, &commit_oids[i]);
+        if (err < 0)
+            goto out;
+    }
+
+    err = (*git_merge_base_xxx)(&oid, self->repo, commit_oid_count, (const git_oid*)commit_oids);
+
+    if (err == GIT_ENOTFOUND)
+        goto out;
```

Comment:
In this case `None` must be returned. But instead it returns `NULL` (return `NULL` means there has been an error, and then an exception must be set).

You can either use `Py_RETURN_NONE` and call `free()` before. Or incref `Py_None` and return that later.

Reply:
Done!